src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/CancelReindexRequestAuthHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using MediatR;
+using Microsoft.Health.Core.Features.Security.Authorization;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Security;
+using Microsoft.Health.Fhir.Core.Messages.Reindex;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.Reindex
+{
+    public class CancelReindexRequestAuthHandler<TCancelReindexRequest, TCancelReindexResponse> : IPipelineBehavior<CancelReindexRequest, CancelReindexResponse>
```

Comment:
>IPipelineBehavior [](http://example.com/codeflow?start=98&length=17)

Why it become pipeline behaviour?

Reply:
I moved most of the logic to the ReindexJobWorker as it may contain a handle to the cancellationToken.  But the authorization piece needs to run first, so auth is checked here before the ReindexJobWorker handles the remainder of the logic.  I could not include the authorization code in the ReindexJobWorker due to initialization dependencies, but this class is only instantiated upon a DELETE request, so initialization will have completed before then.

Reply:
I think it's a bit of overkill to add behaviour.
It just looks strange to have auth seprated only for cancel request?
I would understand if that would serve all reindex requests, but I think it's better to go with abstract class derived from IPipelineHandler<TRequest,TResponse> where TRequest: ReindexRequest, TResponse: ReindexResponse and inherit all other reindex handlers from it.

Moving handler into job worker feels breaking general approach for the sake of optimization for case of one machine, and I'm not sure it's worth it.

Reply:
Agreed, I have simplified the fix to simply check the value in the datastore.